server/builtin/app.go

```diff
@@ -256,3 +264,12 @@ func (a *builtinApp) newLocalizer(creq apps.CallRequest) *i18n.Localizer {
 
 	return a.conf.I18N().GetUserLocalizer(creq.Context.ActingUser.Id)
 }
+
+func (a *builtinApp) checkConfigValid() error {
+	oauthEnabled := a.conf.MattermostConfig().Config().ServiceSettings.EnableOAuthServiceProvider
+	if oauthEnabled == nil || !*oauthEnabled {
+		return errors.Errorf(oauthConfigErrorMessage, a.conf.Get().MattermostSiteURL)
+	}
+
```

Comment:
Should we also check if bot account creation is enabled as in https://github.com/mattermost/mattermost-plugin-apps/blob/52ed70d8eeb68b3c73b41549143736e2065b62db/server/command/service.go#L258-L261 ?

Reply:
@hanzei The apps plugin is now using the plugin API to create bots, which bypasses this config value. Since the config value is now irrelevant in regards to the Apps framework, I've chosen to leave the check out of here.

Reply:
Good point :+1: